src/main/java/io/reactivex/Flowable.java

```diff
@@ -10650,7 +10650,7 @@ public final Completable ignoreElements() {
     @CheckReturnValue
     @BackpressureSupport(BackpressureKind.UNBOUNDED_IN)
     @SchedulerSupport(SchedulerSupport.NONE)
-    public final Maybe<T> reduce(BiFunction<T, T, T> reducer) {
+    public final Maybe<T> reduce(BiFunction<? super T, ? super  T, ? extends T> reducer) {
```

Comment:
This sort of change is a good idea for me but problem is that it is probably a breaking change in that user source that used to compile may no longer compile:

This 
```java
Maybe<Integer> result = flowable.reduce(f);
```
will need to be
```java
Maybe<Integer> result = flowable.<Integer>reduce(f);
```

I haven't checked this yet, can you confirm?

I can live with it. @akarnokd?